test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -582,44 +561,219 @@ public void updateParticipantWithNoAccount() {
+        mockHealthCodeAndAccountRetrieval();
         
-        participantService.updateParticipant(STUDY, ID, PARTICIPANT);
+        participantService.updateParticipant(STUDY, CALLER_ROLES, ID, PARTICIPANT);
         
         verifyNoMoreInteractions(externalIdService);
-        verify(optionsService).setAllOptions(eq(STUDY.getStudyIdentifier()), eq("healthCode"), optionsCaptor.capture());
+        verify(optionsService).setAllOptions(eq(STUDY.getStudyIdentifier()), eq(HEALTH_CODE), optionsCaptor.capture());
         Map<ParticipantOption, String> options = optionsCaptor.getValue();
         assertEquals("POWERS", options.get(EXTERNAL_IDENTIFIER));
     }
     
     @Test(expected = BridgeServiceException.class)
     public void updateParticipantWithNoHealthCode() {
         STUDY.setExternalIdValidationEnabled(true);
-        doReturn(null).when(healthCodeService).getMapping("healthId");
+        doReturn(null).when(healthCodeService).getMapping(HEALTH_ID);
         doReturn(null).when(account).getHealthId();
         doReturn(account).when(accountDao).getAccount(STUDY, ID);
         
-        participantService.updateParticipant(STUDY, ID, PARTICIPANT);
+        participantService.updateParticipant(STUDY, CALLER_ROLES, ID, PARTICIPANT);
     }
     
     @Test
     public void createParticipantWithoutExternalIdAndNoValidation() {
         STUDY.setExternalIdValidationEnabled(false);
-        doReturn(account).when(accountDao).signUp(eq(STUDY), any(), eq(false));
-        doReturn("healthId").when(account).getHealthId();
-        doReturn("id").when(account).getId();
-        doReturn(healthId).when(healthCodeService).getMapping("healthId");
-        doReturn("healthCode").when(healthId).getCode();
-        
+        mockHealthCodeAndAccountRetrieval();
+
         // These are the minimal credentials and they should work.
-        StudyParticipant participant = new StudyParticipant.Builder().withEmail(EMAIL).withPassword("P@ssword1")
-                .build();
-        
-        IdentifierHolder idHolder = participantService.createParticipant(STUDY, participant);
-        assertEquals("id", idHolder.getIdentifier());
+        IdentifierHolder idHolder = participantService.createParticipant(STUDY, CALLER_ROLES, NO_ID_PARTICIPANT);
+        assertEquals(ID, idHolder.getIdentifier());
         verifyNoMoreInteractions(externalIdService); // no ID, no calls to this service
     }
+    
+    @Test
+    public void userCannotSetStatusOnCreate() {
+        verifyStatusCreate(Sets.newHashSet());
+    }
+    
+    @Test
+    public void noRoleCanSetStatusOnCreate() {
+        verifyStatusCreate(Sets.newHashSet(RESEARCHER, ADMIN, DEVELOPER));
+    }
+    
+    @Test
+    public void userCannotChangeStatus() {
+        verifyStatusUpdate(Sets.newHashSet(), null);
+    }
+    
+    @Test
+    public void developerCanChangeStatusOnEdit() {
+        verifyStatusUpdate(Sets.newHashSet(DEVELOPER), AccountStatus.DISABLED);
```

Comment:
Well, we're allowing this because I don't want to write a bunch of complicated logic about who can do what. If you're a study member and you have a user's ID, you can edit them right now. But developers cannot get that information themselves, they'd have to get it from a researcher.